### PR TITLE
[core] Flatten grid packages folder

### PR DESCRIPTION
**Pull request by cherniavskii**
Closes https://github.com/mui/mui-x/issues/10114

### CODE REVIEW

**Review by flaviendelangle — COMMENTED** [review]
You should be able to remove a ternary in the `copyFiles.mjs` file that was due to this nested folder structure

File: scripts/copyFiles.mjs [hunk 1]
```diff
@@ -157,12 +157,7 @@ async function run() {
   try {
     const packageData = await createPackageFile();
 
-    const filesToCopy = [
-      './README.md',
-
-      // TODO: Remove `grid` folder to flatten the structure
-      packageData.name.includes('grid') ? '../../../CHANGELOG.md' : '../../CHANGELOG.md',
-    ];
+    const filesToCopy = ['./README.md', '../../CHANGELOG.md'];
```
Comment (flaviendelangle) [inline scripts/copyFiles.mjs:160]: When I migrated to use the same bundling strategy as the core, this code you just removed was the only diff with the core file
So maybe you could just remove everything from this file and import the core one, like we are doing for `buildTypes.mjs`

But this need to be double checked of course

**Review by romgrk — APPROVED** [review]

File: scripts/copyFiles.mjs (hunk 1 above)
Comment (cherniavskii) [reply scripts/copyFiles.mjs:160]: I'll look into this and submit a follow-up PR if necessary

Comment (cherniavskii) [reply scripts/copyFiles.mjs:160]: Unfortunately, there are other non-trivial differences between our script and the one from the core repo.
Making the `copyFiles` script generic would be a significant effort and I think it's not worth it.
But I think it makes sense to at least reuse some functions from that script.
I'll submit a PR to the core repo.

**Review by MBilalShafi — APPROVED** [review]

**Review by flaviendelangle — APPROVED** [review]

File: scripts/copyFiles.mjs (hunk 1 above)
Comment (cherniavskii) [reply scripts/copyFiles.mjs:160]: https://github.com/mui/material-ui/pull/40970

**Pull request merged**